### DISCUSSION TITLE
Notify on setting identical value

### DISCUSSION THREAD

**Discussion by gismya** [ai/nanostores#276] [Q&A] [answered]
If you want to run the event listener callbacks onSet when the value is the same as the previous value, what's the best way of doing that?

The cleanest way I've come up with now, 
```ts  
onSet($atom, ({ newValue }) => {
    if (newValue === $atom.get()) {
      //@ts-expect-error - notify is not exposed in the types
      $atom.notify(newValue);
    }
  });
```

Would it make sense to expose the notify method in the types, like what's done with value?
Any obvious way to already do it that I'm missing?
Any other feature that could / should be implemented?

**Comment by ai** [comment]
There is a problem with `notify`, it doesn’t work with React.

React and some other frameworks requires change in the value.

The hacky way is to keep value inside the object and change the object:

```js
$atom.set({ value })
```

**Reply by gismya** [reply]
For this part of the code I'm running it in Vanilla javascript, but that's a way to keep in mind if I need to do it inside framework-land.

**Reply by gismya** [reply]
Are there any obvious objections or issues with adding the notify to the types? If not I could work on a PR for that. It would be useful for this, but also for things like making custom lifecycle handlers and similar.

**Reply by ai** [reply]
We removed it because of React issues. We can add it with notice, but is it a good idea?

**Reply by blikblum** [reply]
I have use case for notify. I use web components

**Reply by ai** [reply, marked as answer]
OK, send PR with a notice.

**Reply by gismya** [reply]
@blikblum - it's now merged to the next branch
